src/main/java/seedu/address/model/recipe/RecipeContainsKeywordPredicate.java

```diff
@@ -26,6 +26,20 @@ public boolean test(Recipe recipe) {
                 .anyMatch(keyword -> StringUtil.recipeContainsIgnoreCase(recipe.getSearchSet().toString(), keyword));
     }
 
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < keywords.size(); i++) {
+            if (i == keywords.size() - 1) {
+                sb.append(String.format("\"%s\"", keywords.get(i)));
+            } else {
+                sb.append(String.format("\"%s\", ", keywords.get(i)));
+            }
+        }
+        // this.keywords.forEach(k -> sb.append(String.format("%s ", k)));
+        return sb.toString().trim();
```

Comment:
Maybe might be better to store the stringbuilder as a private variable that's initialised in the constructor? so it'll only be constructed once and simple return sb.toString() in this toString() function

Reply:
the toString() is only really called when theres no matches found so I think this might be better?